jaxmarl/environments/jaxnav/maps/grid_map.py

```diff
@@ -569,6 +651,44 @@ def _checkPointRect(x, y, rx, ry):
         inside = _checkPolyWithinRect(sides, grid_idx[1], grid_idx[0])
         return inside & map_grid[grid_idx[0], grid_idx[1]] & valid_idx
     
+    @partial(jax.jit, static_argnums=[0])
+    def check_all_agent_agent_collisions(self, agent_positions: chex.Array, agent_theta: chex.Array, agent_coords=None) -> chex.Array:
```

Comment:
Also would be good to provide more detail for people (like) me who have no idea what the separating axis theorem is.

Reply:
yea very fair point :)